### PR TITLE
music-assistant: init at 2.0.7

### PR DESCRIPTION
**Pull request by mweinelt**
## Description of changes

https://github.com/music-assistant/server
https://github.com/music-assistant/frontend

and a few other things from https://github.com/music-assistant

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: nixos/modules/services/audio/music-assistant.nix [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+      description = ''
+        List of extra options to pass to the music-assistant executable.
+      '';
+    };
+
+    providers = mkOption {
+      type = listOf (enum cfg.package.providerNames);
+      default = [];
+      example = [
+        "opensubsoic"
```
Comment (mweinelt) [inline nixos/modules/services/audio/music-assistant.nix:55]:
```suggestion
        "opensubsonic"
```

File: pkgs/development/python-modules/py-opensonic/default.nix [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  src = fetchFromGitHub {
+    owner = "khers";
+    repo = "py-opensonic";
+    rev = "v${version}";
+    hash = "sha256-FZJ8njXdSaNuwX9pmFV8Gb7BpTS7Y09h9ZNSjcXkkc8=";
+  };
+
+  build-system = [
+    setuptools
+    wheel
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/py-opensonic/default.nix:23]: Why do we need wheel here? The package is not referring to it except through GHA.

File: nixos/modules/services/audio/music-assistant.nix [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+  meta.buildDocsInSandbox = false;
+
+  options.services.music-assistant = {
+    enable = mkEnableOption "Music Assistant";
+
+    package = mkPackageOption pkgs "music-assistant" { };
+
+    extraOptions = mkOption {
+      type = listOf str;
+      default = [];
```
Comment (SuperSandro2000) [inline nixos/modules/services/audio/music-assistant.nix:41]: We probably want to pass by default --config to not write the data into /var/lib/music-assistant/.musicassistant

```suggestion
      default = [ "--config" "/var/lib/music-assistant" ];
```

**Comment by mweinelt** [comment]
- Scrolling through albums/artists/titles is super slow
  - ```
    ERROR (MainThread) [music_assistant.helpers.process] Detected that you are running the (docker) container without permissive access rights. This will impact performance !
    ```
- They have no browser playback :(

File: pkgs/by-name/mu/music-assistant/package.nix [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+      pythonPath
+      providerPackages
+      providerNames
+    ;
+    tests = nixosTests.music-assistant;
+  };
+
+  meta = with lib; {
+    changelog = "https://github.com/music-assistant/server/releases/tag/${version}";
+    description = "Music Assistant is a music library manager for your offline and online music sources which can easily stream your favourite music to a wide range of supported players";
```
Comment (SuperSandro2000) [inline pkgs/by-name/mu/music-assistant/package.nix:108]:
```suggestion
    description = "Music Assistant is a music library manager for your music which can stream it to a wide range of supported players";
```
I tried to shorten that a bit

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**